qt/python/mantidqtinterfaces/mantidqtinterfaces/dns_powder_tof/data_structures/dns_tof_powder_dataset.py

```diff
@@ -0,0 +1,186 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+
+"""
+Class which loads and stores a single DNS datafile in a dictionary.
+"""
+
+import os
+
+from mantidqtinterfaces.dns_powder_tof.data_structures.object_dict import ObjectDict
+from mantidqtinterfaces.dns_powder_tof.helpers.list_range_converters import \
+    list_to_multirange as list_to_range
+
+
+# copied from dns dataset
+def get_proposal_from_filename(filename, file_number):
+    return filename.replace(f'_{file_number:06d}.d_dat', '')
+
+
+def get_bank_positions(sample_data, rounding_limit=0.05):
+    new_arr = []
+    inside = False
+    banks = set(entry['det_rot'] for entry in sample_data)
+    for bank in banks:
+        for compare in new_arr:
+            if abs(compare - bank) < rounding_limit:
+                inside = True
+                break
+            inside = False
+        if not inside:
+            new_arr.append(bank)
+    return new_arr
+
+
+def _get_max_key_length(dataset):
+    return max([len(a) for a in dataset.keys()]) + 6 + 4
```

Comment:
This method was implemented for alignment purposes inside the generated script, so 4 extra spaces were needed to account for a tab and 6 extra spaces were needed to account for special characters like ', :, ect. Since you requested a modification of the indentation style, this method won't be needed any longer and will be removed on resubmission.